src/TestSuite/LogTrait.php

```diff
@@ -0,0 +1,99 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite;
+
+use Cake\Log\Engine\ArrayLog;
+use Cake\Log\Log;
+use InvalidArgumentException;
+
+/**
+ * Make assertions on logs
+ */
+trait LogTrait
+{
+    /**
+     * @param array|string $engines The engine(s) which should receive a log message
+     * @return void
+     */
+    public function setupLog(array|string $engines): void
+    {
+        Log::reset();
+        if (is_array($engines)) {
+            foreach ($engines as $engine) {
+                Log::setConfig($engine, ['className' => 'Array']);
```

Comment:
Don't we need to allow providing log configs so that one can test for specific levels or scopes for e.g.?